agent/containers/images/image-test-script.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+# A demonstration of how to use pbench-tool-meister-start with an existing
+# Redis server, and one or more remote
+
+read -p "Specify distribution to test: " distrovar
+DISTRO=$distrovar
+
+read -p "Specify image tag to test: " tagvar
+TAG=$tagvar
+
+read -p "Specify redis host: " redishost
+read -p "Specify redis port: " redisport
+export REDIS_HOST=$redishost
+export REDIS_PORT=$redisport
+export PBENCH_REDIS_SERVER="${REDIS_HOST}:${REDIS_PORT}"
+
+function wait_keypress {
+    echo "Press any key to continue"
+    while [[ true ]]; do
+        read -t ${1} -n 1
+        if [[ ${?} = 0 ]]; then
+            return 0
+        else
+            echo "waiting for the keypress"
+        fi
+    done
+}
+
+clear
+
+printf -- "We will now begin the guided pbench image test process. \n\n\n"
+
+wait_keypress 120
+
+# If somebody wants to write their own pbench-tool-meister-start, great, then
+# that person can change the assumption on where the persistent volume is for
+# the Tool Data Sink, and how the "sysinfo", "init", and "end" commands
+# operate with respect to data storage.
+
+export pbench_run=${HOME}/test-run-dir
+rm -rf ${pbench_run}
+mkdir ${pbench_run}
+printf -- "${pbench_run}\n" > ${pbench_run}/.path
+export pbench_log=${pbench_run}/pbench.log
+
+printf -- "\n\nThe benchmark run directory will be found inside:\n\n"
+
+find ${pbench_run} -ls
+
+wait_keypress 120
+
+
+# Register tools
+
+printf -- "\nNow, let's pick a host to collect data from, as well as which tools we wish to register for that host.\n\n"
+
+wait_keypress 120
+
+read -p "Hostname: " hostvar
+echo $hostvar > ${pbench_run}/remotes.lis
+
+
+# Register the specified tools, will be recorded in
+# ${pbench_run}/tools-v1-default
+anothertool=y
+while [ $anothertool == "y" ]
+do
+    read -p "Tool name: " tool
+    printf -- "\npbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis\n"
+    pbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis
+    sleep 1
+    read -p "Another tool? (y/n)" anothertool
+done
+printf -- "\nDone registering tools.\n\n"
+ls -lR ${pbench_run}/tools-v1-default
```

Comment:
I find the switch back and forth between `find` and `ls -lR` a bit disturbing; is there really a reason not to be consistent?

Reply:
Long file name paths.

Reply:
Yes, the format differs; but then, that's one of the reasons that switching back and forth is disruptive. Not a big deal, so I won't dwell on it further.

Reply:
_[Once again unresolving to start a new topic for the same line...]_

Do we really want to rely on finding the tools in `${pbench_run}/tools-v1-default`, or should we be using a command, like `pbench-list-tools`, to generate this output?

Reply:
pbench-list-tools doesn't seem to be able to recognize any tools as registered due to the non-standard registration location.

Reply:
I'm confused...are you saying that the any tools that we registered at lines 149-151 using `pbench-register-tool` cannot be detected at line 153 using `pbench-list-tools`??  How is that?

Reply:
Yeah, it seems to be the case that despite the fact that pbench-register-tools is registering in the new pbench_run dir, pbench-list-tools is still sticking to the default dir, /var/lib/pbench-agent. Even testing it seperately, pbench-list-tools does not change behavior when $pbench_run is defined/changed. I am testing this with a 0.71.0 build that's a few months old. Not sure if it has been addressed since, but may not be a good idea to assume all users have a super recent build.

Reply:
Tracked in Issue #2473.